specification/network/resource-manager/readme.md

```diff
@@ -410,6 +410,41 @@ input-file:
 - Microsoft.Network/preview/2015-05-01-preview/network.json
 ```
 
+## Suppression  
+``` yaml
+directive:
+  - suppress: RequiredPropertiesMissingInResourceModel

```

Comment:
Can we fix it for APIs going forward? So can we suppress it for existing API-versions. And then going forward, new API-versions reference the "resource" model where "id" property is marked as read-only. Would it cause any issues on SDK generation side? @dsgouda ?

Reply:
Please add these suppressions to the specific APis or api-versions and not for the entire thing.

Reply:
@ravbhatnagar  reworked. Please review

Reply:
Hmm... theoretically, yes. Any property not marked as `readonly` could be "set" by a user facing sdk. In this case it would include `id` as well. If existing APIs have some way of detecting an incorrectly set "id" and throw an appropriate error we should be OK. If not, I am not sure what the implications could be.
@fearthecowboy any thoughts here?

Reply:
The constructors for model types would likely change ... so it'd be a binary breaking change.

other than that, the worst that would happen is that someone would change a value, and the server give them an error for trying.

Reply:
So lack of readonly property won't raise serious problems. No breaking changes would be done. Everyone happy?!!

Reply:
IIRC
- We want to enable these suppressions only for the older API versions, maybe we can enable suppressions only for api tags prior to a certain tag, please consult on the offline thread for this since I am not the expert here 
- All new API versions MUST adhere to these rules
- If we merely decide to enable suppressions and not change the spec/api for older versions then yes, there will be no breaking changes introduced, however, newer api versions will definitely cause breaking changes to the subsequently generated sdks.
Hope this makes sense

Reply:
Yes, put the `directive` section with the `suppress` items in the tag'd sections for that specific API version.  Then it will only apply on that version.

Reply:
@dsgouda why do we need these suppressions for older api only? As I understand there is no chance we'll make ids readonly so we'll need to have these exceptions in all the versions... am I missing something?